### PR TITLE
support a/b testing via mixpanel

### PR DESCRIPTION
**Pull request by travisbloom-cb**
closes https://github.com/Stand-With-Crypto/swc-web/issues/143

## What changed? Why?

See the experiments.ts file for commentary/context.

## How has it been tested?

- [X] Locally
- [ ] Vercel Preview Branch
- [ ] Unit test
- [ ] Functional test

## Change management

type=routine 
risk=low 
impact=sev5

### CODE REVIEW

File: src/utils/web/clientLocalUser.ts [hunk 1]
```diff
@@ -42,6 +45,8 @@ export const bootstrapLocalUser = () => {
   getLocalUser()
 }
 
+const log = (msg: string) => logger.info(`getLocalUser - ${msg}`)
```
Comment (lucasrmp-cb) [inline src/utils/web/clientLocalUser.ts:48]: np: doesn't `getLogger` already do this

**Review by lucasrmp-cb — APPROVED** [review]
The typings for experiments config are quite complex, it's smelling a little bit for me (I wasn't able to make them simpler with the same type safety though).
Besides that LGTM

**Comment by travisbloom-cb** [comment]
> The typings for experiments config are quite complex, it's smelling a little bit for me (I wasn't able to make them simpler with the same type safety though). Besides that LGTM

Ya its pretty knarly, but the resulting types are the strongest guarantees I could add

File: src/app/[locale]/internal/user-experiments/page.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  SelectValue,
+} from '@/components/ui/select'
+import { useHasHydrated } from '@/hooks/useHasHydrated'
+import { useLocalUser } from '@/hooks/useLocalUser'
+import { EXPERIMENTS_CONFIG } from '@/utils/shared/experiments'
+import { setLocalUserPersistedValues } from '@/utils/web/clientLocalUser'
+
+export const dynamic = 'error'
+
+export default function ExperimentsHomepage() {
```
Comment (benson-budiman-cb) [inline src/app/[locale]/internal/user-experiments/page.tsx:17]: This internal website is designed to be used by a user to override their own experiment groups, right? To clarify, is it possible to explicitly override the experiment groups for a user that isn't your own?

**Review by paulovictor-cb — APPROVED** [review]
LGTM. A question:

Would it be good to decrease the time that the experiment variant, assigned to a user, is stored? For what I saw, it is stored for a year with the other properties, but imagine the following scenario:

We create an experiment and define, initially, two variants. A week later, we receive the request to add a new variant to this experiment. The users that visited the app during this first week will have their variants defined as one of the first two and won't ever be able to be assigned variant 3.

The problem here that I'm thinking is that if we use the metrics of variants and their related results to define which one is better, is wouldn't be a clear result as variants 1 and 2 would have an advantage over variant 3. 

One solution would be to create a new experiment from scratch using the three variants, or decrease the stored time to try to mitigate this difference for analysis later.

File: src/app/[locale]/internal/user-experiments/page.tsx (hunk 2 above)
Comment (travisbloom-cb) [reply src/app/[locale]/internal/user-experiments/page.tsx:17]: yup. I know product/people will want to see a specific version of an experiment so thats why I added this

**Comment by travisbloom-cb** [comment]
> We create an experiment and define, initially, two variants. A week later, we receive the request to add a new variant to this experiment. The users that visited the app during this first week will have their variants defined as one of the first two and won't ever be able to be assigned variant 3.

Ya, in this scenario we can either 1) change the datetime we view in mixpanel to the point the variant was added. 2) start a new experiment

**Pull request merged**